api/src/main/java/org/openmrs/api/impl/EncounterServiceImpl.java

```diff
@@ -281,19 +281,20 @@ public Encounter getEncounter(Integer encounterId) throws APIException {
 	/**
 	 * @see org.openmrs.api.EncounterService#getEncountersByPatientIdentifier(java.lang.String)
 	 */
+	@Override
 	@Transactional(readOnly = true)
 	public List<Encounter> getEncountersByPatientIdentifier(String identifier) throws APIException {
 		if (identifier == null) {
 			throw new IllegalArgumentException("The 'identifier' parameter is required and cannot be null");
 		}
 		
-		List<Encounter> encs = new Vector<Encounter>();
```

Comment:
What do you base on to tell that this is the cause of the slowness?

Reply:
The way how the search logic is implemented, the method searches actually  more criteria other than just patient identifier, which is not what the name of the method should be doing. And at the same time though the vector class maintains insertion order for the data but it is synchronized thus giving poor performance in searching, adding, delete and update of its elements.

Reply:
Did you in any way test this to confirm that it was slow before your changes but faster afterwards?

Reply:
I tried running against a deidentified copy of my database using a profiler but noticed a less significance.

Reply:
Then what you have done does not fix the reported problem.